src/components/navigation/Navbar.js

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+export default function Navbar(props) {
+  // incoming prop object must be in this format
+  // [{id:'',link:'',url-ext:''},
+  // {id:'',link:'',url-ext:''},
+  // {id:'',link:'',url-ext:''},]
+  //
```

Comment:
This is really nice, and impressive because I see that you want other devs to use your component in a certain way. Great DX leads to higher maintainability of the project.

This is precisely where we can see the power of typescript. If I were to do it in TS, I'd rewrite this as follows
```typescript
type NavbarProps = {
  id: string,
  link: string,
  urlExt: string,
}
export function Navbar(props: NavbarProps) {
...
}
```

Notice how the present code uses `urlExt` but the comment in this component suggests to use `url-ext`. I think there are many such bugs that TS would save you from, much early in the development process.

Reply:
Yeah, I think it's a better way and would be very easy to understand when other developers start working in the project in the future.

I am not using TS right now so I have just changed the comment in the component. Thanks for pointing this out!